### PR TITLE
Add `store_phone_log` to `Profile` mpp 2250

### PR DESCRIPTION
**Pull request by groovecoder**
This PR fixes MPP-2250.

How to test:
1. Check out this code
2. Run `python manage.py migrate`
3. Run `python manage.py runserver`
4. Sign in to http://127.0.0.1:8000
5. Go to http://127.0.0.1:8000/api/v1/profiles/
   * [x] there should be a `store_phone_log` item (defaults to `true`)
   * Note the `id` value
6. Go to http://127.0.0.1:8000/api/v1/docs/
7. Scroll down to the "profiles" endpoints
8. Expand the `PATCH /profiles/{id}/` item
9. Click "Try it out"
10. Change `  "store_phone_log": false,`
11. Enter the `id` value from step 5
12. Click "Execute"
13. Refresh http://127.0.0.1:8000/api/v1/profiles/
    * [x] The `store_phone_log` item should now show `false`

- [x] Commits in this PR are minimal and [have descriptive commit messages](https://chris.beams.io/posts/git-commit/).

### CODE REVIEW

File: emails/migrations/0050_profile_store_phone_log.py [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+            ' ("id" integer NOT NULL PRIMARY KEY AUTOINCREMENT,'
+            ' "api_token" char(32) NOT NULL,'
+            ' "user_id" integer NOT NULL REFERENCES "auth_user" ("id") DEFERRABLE INITIALLY DEFERRED,'
+            ' "address_last_deleted" datetime NULL,'
+            ' "num_address_deleted" integer unsigned NOT NULL CHECK ("num_address_deleted" >= 0),'
+            ' "last_hard_bounce" datetime NULL,'
+            ' "last_soft_bounce" datetime NULL,'
+            ' "subdomain" varchar(63) NULL UNIQUE,'
+            ' "server_storage" bool NOT NULL,'
+            ' "store_phone_log" bool NOT NULL,'
```
Comment (jwhitlock) [inline emails/migrations/0050_profile_store_phone_log.py:30]: Nice job with this messiness! One change, add the database-level default of `true`:

```suggestion
            ' "store_phone_log" bool NOT NULL DEFAULT 1,'
```

File: emails/models.py [hunk 2]
```diff
@@ -126,6 +129,8 @@ def save(self, *args, **kwargs):
         if not self.server_storage:
             relay_addresses = RelayAddress.objects.filter(user=self.user)
             relay_addresses.update(description="", generated_for="", used_on="")
+        # TODO: any time a profile sets store_phone_log to False, delete the
+        # PhoneLog objects for the RelayNumber
```
Comment (jwhitlock) [inline emails/models.py:133]: TODO: Write an issue or Jira ticket to make sure we do this task. Bonus: reference the issue number in the comment.

**Review by jwhitlock — CHANGES_REQUESTED** [review]
Thanks @groovecoder! I didn't notice the comment about PR #2316 building on this one, so I've copied over my comments from that one.

I think the required change is the default in the migration. Thanks for running `black`!

**Review by jwhitlock — APPROVED** [review]

File: emails/models.py (hunk 2 above)
Comment (groovecoder) [reply emails/models.py:133]: https://mozilla-hub.atlassian.net/browse/MPP-2261

**Pull request merged**